ktor-server/ktor-server-host-common/jvmAndNix/src/io/ktor/server/engine/BaseApplicationEngine.kt

```diff
@@ -96,6 +101,24 @@ public abstract class BaseApplicationEngine(
         }
     }
 
+    @OptIn(InternalAPI::class)
+    private fun Application.installDefaultTransformationChecker() {
+        // Respond with "415 Unsupported Media Type" if content cannot be transformed on receive
+        intercept(ApplicationCallPipeline.Plugins) {
+            try {
+                proceed()
+            } catch (e: CannotTransformContentToTypeException) {
+                call.respond(HttpStatusCode.UnsupportedMediaType)
+            }
+        }
+
+        sendPipeline.intercept(ApplicationSendPipeline.After) { subject ->
```

Comment:
It's better to check it earlier, after the `Render` phase. Use `insertPhaseAfter` API here